sdk/compute/Azure.ResourceManager.Compute/tests/Generated/Samples/Sample_AvailabilitySetCollection.cs

```diff
@@ -0,0 +1,246 @@
+            // this example is just showing the usage of "AvailabilitySets_Get" operation, for the dependent resources, they will have to be created separately.
+
+            // authenticate your client
+            ArmClient client = new ArmClient(new DefaultAzureCredential());
+
+            // this example assumes you already have this ResourceGroupResource created on azure
+            // for more information of creating ResourceGroupResource, please refer to the document of ResourceGroupResource
+            string subscriptionId = "{subscription-id}";
+            string resourceGroupName = "rgcompute";
+            ResourceIdentifier resourceGroupResourceId = ResourceGroupResource.CreateResourceIdentifier(subscriptionId, resourceGroupName);
+            ResourceGroupResource resourceGroupResource = client.GetResourceGroupResource(resourceGroupResourceId);
+
+            // get the collection of this AvailabilitySetResource
+            AvailabilitySetCollection collection = resourceGroupResource.GetAvailabilitySets();
+
+            // invoke the operation
+            string availabilitySetName = "aaaaaaaaaaaaaaaaaaaa";
+            bool result = await collection.ExistsAsync(availabilitySetName);
+
+            Console.WriteLine($"Succeeded: {result}");
+        }
+
+        // AvailabilitySets_List_MaximumSet_Gen
+        [NUnit.Framework.Test]
+        [NUnit.Framework.Ignore("Only verifying that the sample builds")]
+        public async Task GetAll_AvailabilitySetsListMaximumSetGen()
+        {
+            // Generated from example definition: specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2022-03-01/examples/availabilitySetExamples/AvailabilitySets_List_MaximumSet_Gen.json
+            // this example is just showing the usage of "AvailabilitySets_List" operation, for the dependent resources, they will have to be created separately.
+
+            // authenticate your client
+            ArmClient client = new ArmClient(new DefaultAzureCredential());
+
+            // this example assumes you already have this ResourceGroupResource created on azure
+            // for more information of creating ResourceGroupResource, please refer to the document of ResourceGroupResource
+            string subscriptionId = "{subscription-id}";
+            string resourceGroupName = "rgcompute";
+            ResourceIdentifier resourceGroupResourceId = ResourceGroupResource.CreateResourceIdentifier(subscriptionId, resourceGroupName);
+            ResourceGroupResource resourceGroupResource = client.GetResourceGroupResource(resourceGroupResourceId);
+
+            // get the collection of this AvailabilitySetResource
+            AvailabilitySetCollection collection = resourceGroupResource.GetAvailabilitySets();
+
+            // invoke the operation and iterate over the result
+            await foreach (AvailabilitySetResource item in collection.GetAllAsync())
```

Comment:
Do we want to show the convenient way to do this with `await foreach (AvailabilitySetResource item in collection)`?

Reply:
Well, I kind of do not want a special case here - because some examples might have optional parameters with values in the example, such like `collection.GetAllAsync(top: 3)`